### DISCUSSION TITLE
Stack safe task array

### DISCUSSION THREAD

**Discussion by florianbepunkt** [gcanti/fp-ts#1563] [Q&A] [answered]
I saw that https://github.com/gcanti/fp-ts/pull/1345 has been merged, but I do not fully understand how to use `sequenceArray`/`sequenceA` or where to find it. I believe the documentation of the pr is outdated.

I have an array of about 10.000 items that are converted. Each conversion requires an api call. The following code works for about 2.000 items, but if the array is larger I get `RangeError: Maximum call stack size exceeded`

How can I make the following stack-safe? I believe `A.sequence(T.ApplicativeSeq)`is the culprit

```ts
pipe(
    paymentTransactions,
    A.map(convertTransaction),
    A.sequence(T.ApplicativeSeq),
    mergeConversionResultsAndErrors,
    T.map(
      flow(
        S.mapLeft(flow(A.map(serializeConversionFailureToCsv), O.fromPredicate(A.isNonEmpty))),
        S.map(flow(A.map(serializePaymentTransactionToDatevCsv), O.fromPredicate(A.isNonEmpty))),
        (r) => ({ erroneousLines: r.left, convertedLines: r.right }),
      ),
    ),
  );
```

EDIT: changing `A.sequence(T.ApplicativeSeq),` to `T.sequenceArray,` does not work, as `T.sequenceArray,` returns a readonly output (while `A.sequence ` does not)

**Comment by DenisFrezzato** [comment, marked as answer]
As you wrote, you have to replace `A.sequence(T.ApplicativeSeq)` with `T.sequenceArray`. If you're not fine with a readonly array, you can use [`toArray`](https://gcanti.github.io/fp-ts/modules/ReadonlyArray.ts.html#toarray) from `ReadonlyArray` module.
